checkmates/data_checks/checks/distribution_data_check.py

```diff
@@ -0,0 +1,155 @@
+
+            >>> X = [5, 7, 8, 9, 10, 11, 12, 15, 20]
+            >>> data_check = DistributionDataCheck()
+            >>> assert data_check.validate(X, y) == [
+            ...     {
+            ...         "message": "Data may have a skewed distribution.",
+            ...         "data_check_name": "DistributionDataCheck",
+            ...         "level": "warning",
+            ...         "code": "SKEWED_DISTRIBUTION",
+            ...         "details": {"distribution type": "positive skew", "Skew Value": 0.7939, "Bimodal Coefficient": 1.0,},
+            ...         "action_options": [
+            ...             {
+            ...                 "code": "TRANSFORM_TARGET",
+            ...                 "data_check_name": "DistributionDataCheck",
+            ...                 "parameters": {},
+            ...                 "metadata": {
+                                    "is_skew": True,
+                                    "transformation_strategy": "yeojohnson",
+            ...                 }
+            ...             }
+            ...         ]
+            ...     }
+            ... ]
+            ...
+            >>> X = pd.Series([1, 1, 1, 2, 2, 3, 4, 4, 5, 5, 5])
+            >>> assert target_check.validate(X, y) == []
+            ...
+            ...
+            >>> X = pd.Series(pd.date_range("1/1/21", periods=10))
+            >>> assert target_check.validate(X, y) == [
+            ...     {
+            ...         "message": "Target is unsupported datetime type. Valid Woodwork logical types include: integer, double, age, age_fractional",
+            ...         "data_check_name": "DistributionDataCheck",
+            ...         "level": "error",
+            ...         "details": {"columns": None, "rows": None, "unsupported_type": "datetime"},
+            ...         "code": "TARGET_UNSUPPORTED_TYPE",
+            ...         "action_options": []
+            ...     }
+            ... ]
+        """
+        messages = []
+
+        if y is None:
+            messages.append(
+                DataCheckError(
+                    message="Data is None",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.TARGET_IS_NONE,
+                    details={},
+                ).to_dict(),
+            )
+            return messages
+
+        y = infer_feature_types(y)
+        allowed_types = [
+            ww.logical_types.Integer.type_string,
+            ww.logical_types.Double.type_string,
+            ww.logical_types.Age.type_string,
+            ww.logical_types.AgeFractional.type_string,
+        ]
+        is_supported_type = y.ww.logical_type.type_string in allowed_types
+
+        if not is_supported_type:
+            messages.append(
+                DataCheckError(
+                    message="Target is unsupported {} type. Valid Woodwork logical types include: {}".format(
```

Comment:
I don't think we need this if this is taken from the `TargetDistributionDataCheck`. Our data check should only run on numerical data, so if a col/target isn't numerical, it should ignore that col

Reply:
Are IntegerNullables supported?